src/librustc/ty/print/mod.rs

```diff
@@ -199,7 +199,7 @@ pub trait Printer<'gcx: 'tcx, 'tcx>: Sized {
             }
         }).count();
 
-        &substs[own_params]
+        SubstsRef::from_slice(self.tcx(), &substs[own_params])
```

Comment:
why does this take `TyCtxt` as an argument? Is there a method that will just pass through the reference if it already has the appropriate lifetime?